src/wallets/hardware/LedgerClient.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { DeepPartial } from "ts-essentials";
+import { mock } from "jest-mock-extended";
+import Transport from "@ledgerhq/hw-transport";
+import TransportWebHID from "@ledgerhq/hw-transport-webhid";
+
+interface CreateLedgerClientParams {
+  client?: DeepPartial<TransportWebHID>;
+  transport?: DeepPartial<Transport>;
+}
+
+const createLedgerClient = (params: CreateLedgerClientParams = {}) => {
+  const client = mock<TransportWebHID>(params.client);
+  const transport = mock<Transport>(params.transport);
+
+  jest.mock("@ledgerhq/hw-transport-webhid", () => {
+    return {
+      ...client,
+      create: () => transport,
+    };
+  });
+
+  const LedgerClient = require("./LedgerClient").default;
+
+  return {
+    client: new LedgerClient(),
+    transport,
+  };
+};
+
+afterEach(() => {
+  jest.resetModules();
+});
+
+describe("getVersion", () => {
+  it("returns the current version", async () => {
+    const { client, transport } = createLedgerClient({
+      transport: {
+        // TODO: Examine real response so this mock is reliable.
```

Comment:
We'll want to have a look at these responses in the real implementation and mimic them. I tend to extract these out as "fixture helpers". The detail of them isn't terribly important usually so I do something like this to keep the tests clean (again don't forget about the real responses - my template on Friday was just a rough approach):

```ts
// ledgerWalletMocks.ts
const createGetVersionResponseMock = () => {
  // TODO: Examine real response so this mock is reliable.
  return Buffer.from([1, 2, 3]);
};

// LedgerClient.spec.ts
const { client, transport } = createLedgerClient({
  transport: {
    send: jest.fn().mockResolvedValue(
      createGetVersionResponseMock()
    ),
  },
});